Allow inline `<cite>` elements

jmsfwk opened the pull request:
Move the block styling and reduced font size on `<cite>` to only
apply when it is a child of `<blockquote>`.

Fixes issue #10226

jmsfwk committed:
Allow inline `<cite>` elements

Move the block styling and reduced font size on `<cite>` to only
apply when it is a child of `<blockquote>`.

Fixes issue #10226

kball commented:
This makes sense to me, but is a breaking change... can we wrap up the block form of citation into a mixin so that if someone wants to re-instate the old behavior it is as simple as 
```
cite {
  @include block-citation();
}
```

jmsfwk committed:
Add  mixin to restore old behavior

ncoden commented:
Hey @jmsfwk. Thank you, I see you made the requested changes, and sorry for the late reply. I think this PR goes to the right direction, which is trying to not impose any design and simply providing a set of easily customizable tools.

It's sad I was not active when you opened this PR, I would recommand a more "component" approach rather than "allowing a old-behavior reset". Anyway, I'll take care of the changes and merge this right after.

Sorry again for the late reply.

ncoden committed:
feat: make cite-block an optional component & use it as default for <cite>

Note: this remove the backward incompability included in 18af0ed805ebe7a18a1b77dd09cd2e1a33a5b469

ncoden commented:
@jmsfwk For your future pull requests, you can create a dedicated branch (like feat/optional-block-cite) and create a pull request from it. So you can continue working on your branch develop while this PR is open.

ncoden commented:
I pushed.
Now we have a `.cite-block` component, generated from its `cite-block()` mixin, and used as default (or not) for the `<cite>` tag according to the `$enable-cite-block` property.

So no migration is needed anymore, and we're slowly moving to a "component" approach for typography.

ncoden committed:
docs: add docs about the `cite-block` component
